site/docs/expressions/specialized_record_expressions.md

```diff
@@ -14,7 +14,7 @@ These expressions allow structs, lists, and maps to be constructed from a set of
 ## Cast Expression
 To convert a value from one type to another, Substrait defines a cast expression. Cast expressions declare an expected type, an input argument and an enumeration specifying failure behavior, indicating whether cast should return null on failure or throw an exception.
 
-Note that Substrait always requires a cast expression whenever the current type is not exactly equal to (one of) the expected types. For example, it is illegal to directly pass a value of type `i8[0]` to a function that only supports an `i8?[0]` argument, some other variation of `i8`, or `i16[0]`.
+Note that Substrait always requires a cast expression whenever the current type is not exactly equal to (one of) the expected types.
```

Comment:
Well, I did kind of like the i8 / i8? example.  This point is kind of subtle and so I believe the example was very intentionally placed.  It was only the discussion of "variants" (which is a different concept from nullability) that I thought was incorrect.

Reply:
Added back! Sorry, I had misunderstood your original comment.